packages/text-field/Input.js

```diff
@@ -27,13 +27,10 @@ import classnames from 'classnames';
 import {VALIDATION_ATTR_WHITELIST} from '@material/textfield/constants';
 
 export default class Input extends React.Component {
-  constructor(props) {
-    super(props);
-    this.inputElement = React.createRef();
-  }
+  inputElement = React.createRef();
+  state = {disableProgramaticSetValue: false};
```

Comment:
Change this to `enableProgrammaticSetValue`, it's easier to reason about logic if the variable is positive